### PR TITLE
Handle any method as mock request method

### PR DESCRIPTION
**Pull request by piotrekfus91**

### CODE REVIEW

File: mockserver/src/main/groovy/pl/touk/mockserver/server/Mock.groovy [hunk 1]
```diff
@@ -60,7 +60,7 @@ class Mock implements Comparable<Mock> {
 
     boolean match(Method method, MockRequest request) {
         boolean usesCondition = hasLimitedUses() ? usesLeft > 0 : true
-        return usesCondition && this.method == method && predicate(request)
+        return usesCondition && (this.method == method || this.method == Method.ANY) && predicate(request)
```
Comment (alien11689) [inline mockserver/src/main/groovy/pl/touk/mockserver/server/Mock.groovy]: Let's add some documentation for this behaviour and option

File: mockserver-api/src/main/xsd/pl/touk/mockserver/api/common.xsd [hunk 2]
```diff
@@ -11,6 +11,7 @@
             <xs:enumeration value="HEAD"/>
             <xs:enumeration value="OPTIONS"/>
             <xs:enumeration value="PATCH"/>
+            <xs:enumeration value="ANY" />
```
Comment (alien11689) [inline mockserver-api/src/main/xsd/pl/touk/mockserver/api/common.xsd]: HTTP does not change very often, but maybe we should name it in safe way to be sure that it won't clash with new things e.g. `_ANY`?

**Review by alien11689 — CHANGES_REQUESTED** [review]

File: mockserver/src/main/groovy/pl/touk/mockserver/server/Mock.groovy (hunk 1 above)
Comment (piotrekfus91) [reply mockserver/src/main/groovy/pl/touk/mockserver/server/Mock.groovy]: Done.

File: mockserver-api/src/main/xsd/pl/touk/mockserver/api/common.xsd (hunk 2 above)
Comment (piotrekfus91) [reply mockserver-api/src/main/xsd/pl/touk/mockserver/api/common.xsd]: Done.

**Review by alien11689 — APPROVED** [review]

**Pull request merged**